rust-lib/src/lib.rs

```diff
@@ -346,3 +355,61 @@ pub extern "C" fn transfer(
     )
     .into()
 }
+
+fn _accept(
+    runtime: &COpaqueStruct,
+    consignment_bytes: *mut c_char,
+    reveal_outpoints: *mut c_char,
+) -> Result<(), RequestError> {
+    let runtime = Runtime::from_opaque(runtime)?;
+
+    let c_consignment_bytes = unsafe { CStr::from_ptr(consignment_bytes) };
+    let consignment_bytes: &[u8] = c_consignment_bytes.to_bytes();
```

Comment:
Why we are doing this via `CStr`? We need just `void *` type (sorry if do not undertand something, never was working with C/Rust interoperability layer yet)

Reply:
I'm also not expert at all. I tried to get the parameter as a `c_void`, but, from my understanding, then we would need to receive (from the sdk user) also the length of the bytes array, or use an opaque struct. Do you want me to investigate more into the matter or should we open an issue for later improvement?

Reply:
I am afraid we need investigation, since c strings are 0-byte terminated and binary consignment object (frequently containing zeros inside) will be probably cut here, leading to a hardly-trackable bugs

Reply:
after a lot of studying and headbanging on the table, I've reached some interesting results :tada:

soon I'll clean the code and push it, so you (@dr-orlovsky) can tell me if you like the solution. unfortunately swig doesn't handle arrays automatically, so I had to construct them in node.js using some helper functions that I added to the swig interface (will need to do that also for other language bindings).

Reply:
pushed. to recap:
- on the rust library I've switched the from `c_char` to `u8` and add a `consignment_length` parameter
- I added some helper methods to the swig interface for node.js that allow constructing c++ arrays from js code
- I added a middleware interface in js (`ffi/nodejs/rgb_node.js`), which has the purpose of hiding some complexity to the sdk user
- I changed the node.js demo in order to load the middleware interface (also fixed/improved minor stuff related to js)

if you agree with this solution, we'll need to change also the swig interface for android. on the contrary, nothing to do for iOS (I think, I still cannot try the demo for that issue with libzmq I mentioned in IRC).

currently the call to `strict_decode` returns a `StrictEncoding(Io(UnexpectedEof))` error, but the logged `consignment_bytes` seems to match correctly the bytes of the consignment file (retrived by `hd`). @dr-orlovsky  do you have any idea on why this error could arise?